File: common/src/main/java/io/github/candycalc/util/math.java
package io.github.candycalc.util;

import net.minecraft.util.Mth;

public class math {
    // Bottosson's linear sRGB blend as per Skye's recommendation
    public static int color_lerp(float m, int x, int y) {
        return (int) (255 * sRGB(Mth.lerp(m, inv_sRGB((float) x / 255), inv_sRGB((float) y / 255))));
    }

    public static double sRGB(double x) {
        if (x >= 0.0031308) {
            return ((1.055) * Math.pow(x, 1.0 / 2.4) - 0.055);
        } else {
            return 12.92 * x;
        }
    }

    public static double inv_sRGB(double x) {
        if (x >= 0.04045)
            return Math.pow((x + 0.055)/(1 + 0.055), 2.4);
        else
            return x / 12.92;
    }
}

File: src/main/kotlin/io/github/candycalc/util/math.kt
package io.github.candycalc.util

import net.minecraft.util.math.MathHelper
import kotlin.math.pow

class math {
    companion object {
        // Bottosson's linear sRGB blend as per Skye's recommendation
        fun color_lerp(m: Float, x: Int, y: Int): Int {
            return (255 * sRGB(
                MathHelper.lerp(
                    m,
                    inv_sRGB((x.toFloat() / 255)),
                    inv_sRGB((y.toFloat() / 255))
                )
            )).toInt()
        }

        fun sRGB(x: Float): Float {
            return if (x >= 0.0031308f) ((1.055f) * x.pow(1.0f / 2.4f) - 0.055f) else 12.92f * x
        }

        fun inv_sRGB(x: Float): Float {
            return if (x >= 0.04045f) ((x + 0.055f) / (1f + 0.055f)).pow(2.4f) else x / 12.92f
        }
    }
}